cargo-marker/src/backend.rs

```diff
@@ -111,7 +111,12 @@ pub fn run_check(config: &Config, info: CheckInfo, additional_cargo_args: &[Stri
 pub fn to_marker_lint_crates_env(lints: &[LintCrate]) -> OsString {
     let lint_paths: Vec<_> = lints
         .iter()
-        .map(|krate| OsString::from(krate.file.as_os_str()))
+        .map(|krate| {
+            let mut env_str = OsString::from(&krate.name);
+            env_str.push(":");
+            env_str.push(OsString::from(krate.file.as_os_str()));
```

Comment:
This felt bad, I'm not super familiar with working with OsString, is there a better way to do this?

Reply:
It should be sufficient to just have the `krate.file.as_os_str()` like this:

```suggestion
            env_str.push(":");
            env_str.push(krate.file.as_os_str());
```

In a previous PR we've discussed, that we can move over to assume that all paths will be valid UTF8. So if you ever get stuck on some `OsString` stuff, you can see if it would work better with normal Strings :)